spec/integration/hawkular_client_spec.rb

```diff
@@ -258,13 +312,13 @@ module Hawkular::Client::RSpec
               actual_data[:data] = data
             end
             on.failure do |error|
-              actual_data[:data] = {}
-              puts 'error callback was called, reason: ' + error.to_s
+              actual_data[:data] = error
             end
           end
 
           actual_data = wait_for actual_data
-          expect(actual_data['status']).to eq('OK')
+          expect(actual_data['status']).to eq('OK') unless @agent_immutable
```

Comment:
I don't like this conditional on the test, immutable agents should be tested on a different test case.